app/controllers/application_controller.rb

```diff
@@ -173,7 +173,7 @@ def after_sign_in_path_for(resource)
 
       uri = Addressable::URI.parse(path)
       uri.query_values = if uri.query_values
-                           uri.query_values.merge(signin_param)
+                           uri.query_values = signin_param
```

Comment:
Parsing the path using `Addressable::URI` was adding `i=i` (internal navigation) into the redirect path after OAuth callback is successful. We only need the signin_param (`{ "signin" => "true" }`) in this case so assignment works well for us

Reply:
Looking at the [failing Cypress test](https://github.com/forem/forem/blob/732f200b4bb91dc6288051ac007daa7df5fb8349/cypress/e2e/seededFlows/loginFlows/userLogin.spec.js#L37) there seems to be an issue with the `prefill` param that does require a small tweak here. Now looking into it